### DISCUSSION TITLE
Would it be possible to use custom middleware on action routes (filament.exports.download) ?

### DISCUSSION THREAD

**Discussion by electronick86** [filamentphp/filament#11365] [Help]
### Package

Actions

### Package Version

v3.2.2

### How can we help you?

Hello.

I would like to use the export action. I have a problem to download the files because my users are not authenticated. I should use another middleware than `web` and `auth` to get my user authenticated.

Would you see a way to achieve this? 

The two middlewares are registered here : 

```php 
//vendor/filament/actions/routes/web.php
Route::get('/filament/exports/{export}/download', DownloadExport::class)
    ->name('filament.exports.download')
    ->middleware(['web', 'auth']);
```

Thanks in advance for ideas!

**Comment by reidsolon** [comment]
What I did is I override the middleware group in my AppServiceProvider

```
// AppServiceProvider.php

use Illuminate\Routing\Router;

public function register(): void
{
    ...
    app(Router::class)->middlewareGroup('filament.actions', ['your', 'custom', 'registered', 'middlewares']);
    ...
}
```
